### ISSUE TITLE
Can't read v1beta1 pipelineruns logs

### ISSUE THREAD

**Issue by deepio** [tektoncd/cli#1865]
# Versions and Operating System

- Kubernetes version: 1.25.4

  **Output of `kubectl version`:**

```bash
#kubectl version --output=yaml
clientVersion:
  buildDate: "2022-05-24T12:17:11Z"
  compiler: gc
  gitCommit: 3ddd0f45aa91e2f30c70734b175631bec5b5825a
  gitTreeState: clean
  gitVersion: v1.24.1
  goVersion: go1.18.2
  major: "1"
  minor: "24"
  platform: darwin/amd64
kustomizeVersion: v4.5.4
serverVersion:
  buildDate: "2022-12-16T19:44:08Z"
  compiler: gc
  gitCommit: b969368e201e1f09440892d03007c62e791091f8
  gitTreeState: clean
  gitVersion: v1.25.4
  goVersion: go1.19.3
  major: "1"
  minor: "25"
  platform: linux/amd64
```

- Tekton Pipeline version:

  **Output of `tkn version` or `kubectl get pods -n tekton-pipelines -l app=tekton-pipelines-controller -o=jsonpath='{.items[0].metadata.labels.version}'`**

```v0.43.2```

- Operating System:

```macos 12.5.1```

# Expected Behavior

Logs

# Actual Behavior

```
W0125 12:45:25.473679   35773 reflector.go:347] github.com/tektoncd/pipeline/pkg/client/informers/externalversions/factory.go:117: watch of *v1.PipelineRun ended with: an error on the server ("unable to decode an event from the watch stream: unable to decode watch event: no kind \"PipelineRun\" is registered for version \"tekton.dev/v1\" in scheme \"github.com/tektoncd/pipeline/pkg/client/clientset/versioned/scheme/register.go:32\"") has prevented the request from succeeding
```

# Steps to Reproduce the Problem

1. Don't use `tekton.dev/v1` because you can't specify serviceAccountName https://tekton.dev/docs/pipelines/pipelineruns/#configuring-a-pipelinerun
```bash
# When you do use v1, you get this.
admission webhook "webhook.pipeline.tekton.dev" denied the request: mutation failed: cannot decode incoming new object: json: unknown field "serviceAccountName"
```
2. Apply a pipelinerun: `kubectl apply -f test.yaml`
```yaml
---
apiVersion: tekton.dev/v1beta1
kind: PipelineRun
metadata:
  name: "testing-stuff"
  namespace: <snip>
  labels:
    app.kubernetes.io/name: python
    app.kubernetes.io/component: pipelinerun
    app.kubernetes.io/part-of: <snip>
    app.kubernetes.io/managed-by: <snip>
spec:
  serviceAccountName: <snip>
  pipelineRef:
    name: run-python-tests
  timeouts:
    pipeline: "3h0m0s"
  params:
    <snip>

  workspaces:
    - name: codebase
      volumeClaimTemplate:
        metadata:
          name: testing-stuff"
        spec:
          accessModes:
            - ReadWriteOnce
          resources:
            requests:
              storage: 15Gi
```
3. If you ask for the logs using something like k9s or lens, you can see them and the build is running.
4. `kubectl logs --selector=tekton.dev/pipelineRun=testing-stuff -n harbormaster --all-containers` doesn't give logs for the tasks, only the pipelinerun? Must use `selector=tekton.dev/taskRun` to get the task logs?

# Additional Info
This may be a problem for both pipelines and the cli

**Comment by deepio** [comment]
Actually I think this is a pipeline issue.

**Comment by deepio** [comment]
On second though, no it's cli because the task is running.

**Comment by lbernick** [comment]
@deepio FYI you can specify a service account for a v1 pipelinerun using `pipelinerun.spec.taskRunTemplate.serviceAccountName`

**Comment by deepio** [comment]
@lbernick that's not important because I can continue to use the beta resource. The real problem is inconsistent logging as-per the title. If there is no log then there's no proof of what happened and it's all for naught. Tkn has problems retrieving logs on steps that occur on or after retries when they are clearly visible in other manual applications such as lens, or k9s.

**Comment by lbernick** [comment]
Thanks, I should have read the issue in more detail. This looks like a duplicate of https://github.com/tektoncd/cli/issues/1847 which has been addressed on `main` (although I am not sure if it has been released yet)

**Comment by piyush-garg** [comment]
PR added
 - https://github.com/tektoncd/cli/pull/1960
 - https://github.com/tektoncd/cli/pull/1961
 
 This should be fixed with 0.30.0 release
